persistence/commons/src/main/java/org/apache/causeway/persistence/commons/integration/repository/RepositoryServiceDefault.java

```diff
@@ -116,14 +119,24 @@ public <T> T persist(final T domainObject) {
         return domainObject;
     }
 
-
     @Override
     public <T> T persistAndFlush(final T object) {
         persist(object);
-        transactionService.flushTransaction();
+        if(!threadLocal.get().getOrDefault(object.getClass(), Boolean.FALSE)) {
+            transactionService.flushTransaction();
+        }
         return object;
     }
 
+    @Override
+    public <T extends Class> void setBulkMode(final T aClass, final Boolean bulkMode) {
```

Comment:
See previous comment, instead of this setter we use an `execInBulk(Callable)` action.

Reply:
Fixed